sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/EliminateJoinToEmptyRelation.scala

```diff
@@ -53,5 +57,23 @@ object EliminateJoinToEmptyRelation extends Rule[LogicalPlan] {
 
     case j @ Join(_, _, LeftSemi, _, _) if canEliminate(j.right, EmptyHashedRelation) =>
       LocalRelation(j.output, data = Seq.empty, isStreaming = j.isStreaming)
+
+    case j @ Join(_, _, LeftAnti, None, _) =>
+      val isNonEmptyBroadcastedRightSide = j.right match {
+        case LogicalQueryStage(_, stage: BroadcastQueryStageExec)
```

Comment:
nit:
```
case LogicalQueryStage(_, stage: QueryStageExec) =>
  stage.getRuntimeStatistics.rowCount.exists(_ > 0)
```

Reply:
@cloud-fan - thanks, use row count is better, updated.